### DISCUSSION TITLE
gn maybe replace msvc flag with something gcc understands ?

### DISCUSSION THREAD

**Discussion by revelator** [msys2/MINGW-packages#15034] [Ideas]
003-remove-msvc-flags.patch has out_ << " /DEF:"; replaced by empty command but would it not make more sense to use out_ << " -Wl,--out-implib,"; which is the gcc equivalent ?.

**Comment by raedrizqie** [comment]
No.. that flag is just used to parse .def file when linking dll

**Reply by revelator** [reply]
uh yeah exactly my point /DEF: does the same for msvc so i was wondering.

**Reply by raedrizqie** [reply]
I dont think there is a special flag for gcc to parse def files when linking, is there any? It will be treated as regular source files..

**Reply by mmuetzel** [reply]
GCC doesn't need an input argument to specify .def files. They are passed as-is like any other source file.
The parameter `-Wl,--out-implib,` that you propose is used to specify the name of the import library that the linker should create. That isn't (and must not be) the name of the .def file.

**Comment by Biswa96** [comment]
`-Wl,--out-implib` is used with the import library to generate it. The `/DEF:` option is used with .def file to export from DLLs.
For example, `-Wl,--out-implib` is present in mingw-w64-angleproject/001-add-mingw-toolchain.patch.

**Reply by revelator** [reply]
well there also is an export version version that is also why i was a bit confused that it needed to be empty. I could be wrong that is why im asking any who have some better understanding of it :)

**Reply by revelator** [reply]
atleast with codeblocks when creating an import library a .def file is allways created, not sure if it is ever used or if it is just created for convinience ?. but then again import libraries are pretty much convinience as well as it is quite possible to link to the dll itself.

**Reply by mmuetzel** [reply]
IIUC, the .def file isn't created in this case. It is used as an **input**.

**Reply by revelator** [reply]
ok
